packages/rest/test/unit/rest-server/rest-server.test.ts

```diff
@@ -1,36 +1,38 @@
 // Copyright IBM Corp. 2013,2017. All Rights Reserved.
-// Node module: @loopback/core
+// Node module: @loopback/rest
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
 import {expect, ShotRequest} from '@loopback/testlab';
 import {anOperationSpec} from '@loopback/openapi-spec-builder';
 import {Binding, Context} from '@loopback/context';
+import {Application} from '@loopback/core';
 import {
-  Application,
+  RestServer,
   ServerRequest,
   BindElement,
   GetFromContext,
   Route,
   InvokeMethod,
-  CoreBindings,
+  RestBindings,
 } from '../../..';
 
-const SequenceActions = CoreBindings.SequenceActions;
+const SequenceActions = RestBindings.SequenceActions;
 
 describe('Application', () => {
   describe('"logError" binding', () => {
     it('provides a default', async () => {
       const app = new Application();
-      const logError = await app.get(SequenceActions.LOG_ERROR);
+      const server = new RestServer(app, {});
+      const logError = await server.get(SequenceActions.LOG_ERROR);
       expect(logError.length).to.equal(3); // (err, statusCode, request)
     });
 
     // tslint:disable-next-line:max-line-length
-    it('can be customized by overriding Application._logError() method', async () => {
+    it('can be customized by overriding RestServer._logError() method', async () => {
       let lastLog: string = 'logError() was not called';
 
-      class MyApp extends Application {
+      class MyServer extends RestServer {
```

Comment:
Hmm, this is adding more friction than I'd like. While I think people should be creating their own application subclasses (where it's easy to customize `_logError` function), I find it weird that one have to subclass `RestServer` too. It will also change the binding key used by the server (`MyServer` is bound to `servers.MyServer`, not `server.RestServer`).

Before, users have several options how to change the error logger:

 1. subclass `Application` and override the method
 2. `app.bind(SequenceActions.LOG_ERROR).to(myErrorLogger)`
 3. (hacky) set the method directly on `Application` instance (works in JS, does not work in TS)

The first option requires a subclass of `RestServer` now, I find that as too much work.

I think the second option no longer works - the default sequence action bound in server-local context will override the custom implementation bound at application level.

The third ways remains as easy to use as before, but it's not a viable solution for people writing in TypeScript.

**We need to give more thoughts to this part and try to find a solution that will be easy to use and work both in TS and JS.**

Reply:
> I think the second option no longer works - the default sequence action bound in server-local context will override the custom implementation bound at application level.

BTW I think this is true for all sequence actions, not only `LOG_ERROR`. We need to approach this from the point of view of application developers and make sure the process for customizing sequence actions is super easy.

Reply:
How about binding REST-specific sequence actions in the Application context? Something along the following lines:

```ts
class Server {
  constructor(app: Application, ...) {
    app.bind(RestBindings.LOG_ERROR).to(...);
  }
}
```

Perhaps even better, can we change this REST transport into a component that's registered via standard component registration? Then the rest component can simply list providers that should be registered in the application context.

```ts
class RestComponent {
  providers: {
    [RestBindings.LOG_ERROR]: LogErrorProvider,
    // ...
  }
}

const app = new Application({
  components: [RestComponent],
  // ...
}
```

Thoughts?